pkg/apis/componentconfig/types.go

```diff
@@ -62,3 +62,213 @@ const (
 	ProxyModeUserspace ProxyMode = "userspace"
 	ProxyModeIPTables  ProxyMode = "iptables"
 )
+
+type KubeletConfiguration struct {
```

Comment:
Defaults will be moved to https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/componentconfig/v1alpha1/defaults.go and will be applied like the defaulting of any other api object. This is not done in the current PR because this PR does not introduce the external type (this will be in a follow up)